arch/arc/core/regular_irq.S

```diff
@@ -35,6 +35,168 @@ GTEXT(_rirq_common_interrupt_swap)
  * TODO: Revist this if FIRQ becomes configurable.
  */
 
+/*
+
+===========================================================
+ RETURN FROM INTERRUPT TO COOPERATIVE THREAD
+===========================================================
+
+That's a special case because:
+ 1. We return from IRQ handler to a cooperative thread
+ 2. During IRQ handling context switch did happen
+ 3. Thread we're returining to previously voluntarily gave control
```

Comment:
some typo's, suggest to use : ... "returning to a thread which previously gave control"

Reply:
Ok, good. Will change that.